### ISSUE TITLE
Can't save charts without closing GC.

### ISSUE THREAD

**Issue by DougLeonard** [GoldenCheetah/GoldenCheetah#2544]
Build 3962 3.5-DEV1705
There is no obvious way to save charts while editing other than closing GC, and crashes lose edits.  Thus when debugging a crash, it seems one needs to exit and restart GC often, export, or otherwise copy edited text.

Ideal behavior: cntl-s saves.

**Comment by liversedge** [comment]
unlikely to do this as the underlying logic makes no sense:

if a chart causes GC to crash then making it permanent would cause GC to crash on startup.

**Comment by DougLeonard** [comment]
It's a good point.  Loosing an hour's work on some coding (that was working until some particular change) is also a point though.  Your scenario can still happen anyway, if you exit before refreshing.  My solution to your point was hitting esc quickly when restarting, but maybe quickly enough won't always be possible.  Maybe a more fine-grained solution could work.  This is not an ideal situation for a code editor.

**Comment by DougLeonard** [comment]
I don't think worrying about crash is a good reason to not let people save work.  Work is valuable.  There should always be a way to save and then access the work, crash or not, bugs or not.  My solution simply isn't complete.  

Crashes in the chart aren't even the only thing that may cause this.  Crashes elsewhere in the program or in the OS can cause loss of work.  Work should always be savable.

**Comment by liversedge** [comment]
Charts are saved on exit.

**Comment by DougLeonard** [comment]
It's your code, but closing an app is usually associated with not saving.

**Comment by liversedge** [comment]
You don't need to save, if the code causes a crash it shouldn't be saved. 
If you are really desperate to save then just export the chart.

**Comment by DougLeonard** [comment]
Desparate? I guess if I have a bug in an hours work I should just start
over and be grateful it doesn't bootlup GC. Anyway, yes export can work,
with 4 mouse clicks.
I actually had no idea if it was saving until the first time I restarted,
and still didn't know when it was saving, until it crashed and I realized
it didn't. At that point I thought either cntrl-s was doing the saving or
that it was periodic, so yeah, I was surprised to loose work.  Not obvious
or predictable. But yes, export can work.
